src/Umbraco.Cms.Integrations.Search.Algolia/App_Plugins/UmbracoCms.Integrations/Search/Algolia/js/algolia.service.js

```diff
@@ -1,17 +1,32 @@
 ﻿function algoliaService(contentTypeResource) {
     return {
         getContentTypes: function (callback) {
+
             contentTypeResource.getAll().then(function (data) {
-                callback(data.filter(item => item.parentId == -1 && !item.isElement).map((item) => {
-                    return {
-                        id: item.id,
-                        icon: item.icon,
-                        alias: item.alias,
-                        name: item.name,
-                        selected: false,
-                        allowRemove: false
+
+                var contentTypesArr = [];
+
+                for (let i = 0; i < data.length; i++) {
+                    
+                    if (data[i].isElement) continue;
+
+                    contentTypeResource.getWhereCompositionIsUsedInContentTypes(data[i].id).then(function (response) {
```

Comment:
I'm wondering if it's actually correct to do this check.  In most cases it probably is, but you could have for example an "Article Page" and then a "Special Article Page", that composes the former and adds a few properties.  But they are both valid pages for editors to create.

So perhaps just filter our the elements and that'll do here.